src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -58,6 +58,37 @@ public override void Initialize (AnalysisContext context)
 				var invocationOperation = (IInvocationOperation) context.Operation;
 				ProcessInvocationOperation (context, invocationOperation);
 			}, OperationKind.Invocation);
+
+			context.RegisterSymbolAction (context => {
+				ProcessGenericTypes (context);
+			}, SymbolKind.NamedType);
+		}
+
+		static void ProcessGenericTypes (SymbolAnalysisContext context)
+		{
+			// Check the Symbol to see the generics relationship, specifically around DAMT
+			// We will initially explore if an unbounded generic type has mismatched annotations with its base-class,
+			// which is an open generic type. i.e. Bar<T> : Foo<[DAMT] T>
+
+			INamedTypeSymbol derivedType = (INamedTypeSymbol) context.Symbol;
```

Comment:
This should check for RUC first, in which case is not worth analyzing because we cannot produce the warning

Reply:
Thanks!